pkg/dnsutil/hostsstore/hostsstore.go

```diff
@@ -195,6 +196,16 @@ func (x *hostsStore) DeallocHostsFile(id string) (err error) {
 	return x.safeStore.WithLock(func() error { return x.safeStore.Delete(id, hostsFile) })
 }
 
+func (x *hostsStore) DeallocHostsDir(id string) (err error) {
```

Comment:
@haytok 

Suggesting that:
- we call this method `Delete`, in line with the `safestore.Store` implementation.
- we remove the `DeallocHostsFile` method, and replace the one call to it (which is inside `container.Remove`), by a call to this new `Delete` method

The point being: since we are cleaning up these directories on `failed create`, we should do the same on `remove` now.

LMK what you think.

Reply:
I was thinking the same thing.
The logic to remove `/var/lib/nerdctl/1935db59/etchosts/<namespace>/<container id>/hosts` is called at the following location within `container.Remove()`.

- https://github.com/containerd/nerdctl/blob/main/pkg/cmd/container/remove.go#L222-L228

The cleanup process of `/var/lib/nerdctl/1935db59/etchosts/<namespace>/<container id>`, which is called when container creation fails, is also needed to be  called within `container.Remove()`.

So, when this PR is merged, is it OK to rewrite `hs.DeallocHostsFile(id)` in `container.Remove()` to `hs.DeallocHostsDir(id)` in another PR?

Reply:
They are definitely related changes, so, not shocking to have them together here in this PR.
Or in another PR.
Whatever you are most comfortable with.
:-)

Reply:
> we call this method Delete, in line with the safestore.Store implementation.
we remove the DeallocHostsFile method, and replace the one call to it (which is inside container.Remove), by a call to this new Delete method

I agree with [apostasie](https://github.com/apostasie) that `pkg/dnsutil/hostsstore/hostsstore.go` should be rewritten as follows to implement calling `hs.Delete(id)` within `container.Create()` and `container.Remove()`.

<details><summary>Details</summary>
<p>

```bash
 /Users/haytok/workspace/nerdctl [fix-to-clean-up-orphaned-files]
> gd pkg/dnsutil/hostsstore/hostsstore.go
diff --git a/pkg/dnsutil/hostsstore/hostsstore.go b/pkg/dnsutil/hostsstore/hostsstore.go
index cf691542..507f93ec 100644
--- a/pkg/dnsutil/hostsstore/hostsstore.go
+++ b/pkg/dnsutil/hostsstore/hostsstore.go
@@ -89,8 +89,7 @@ type Store interface {
        Release(id string) error
        Update(id, newName string) error
        HostsPath(id string) (location string, err error)
-       DeallocHostsFile(id string) (err error)
-       DeallocHostsDir(id string) (err error)
+       Delete(id string) (err error)
        AllocHostsFile(id string, content []byte) (location string, err error)
 }

@@ -186,24 +185,13 @@ func (x *hostsStore) AllocHostsFile(id string, content []byte) (location string,
        return x.safeStore.Location(id, hostsFile)
 }

-func (x *hostsStore) DeallocHostsFile(id string) (err error) {
-       defer func() {
-               if err != nil {
-                       err = errors.Join(ErrHostsStore, err)
-               }
-       }()
-
-       return x.safeStore.WithLock(func() error { return x.safeStore.Delete(id, hostsFile) })
-}
-
-func (x *hostsStore) DeallocHostsDir(id string) (err error) {
-       defer func() {
-               if err != nil {
-                       err = errors.Join(ErrHostsStore, err)
-               }
-       }()
+func (x *hostsStore) Delete(id string) (err error) {
+       err = x.safeStore.WithLock(func() error { return x.safeStore.Delete(id) })
+       if err != nil {
+               err = errors.Join(ErrHostsStore, err)
+       }

-       return x.safeStore.WithLock(func() error { return x.safeStore.Delete(id) })
+       return err
 }

 func (x *hostsStore) HostsPath(id string) (location string, err error) {
```

</p>
</details>

Reply:
Fixed